Align headings in org-variable-pitch

terlar opened the pull request:
I noticed that when I am scaling org-level faces that the headings won't be aligned with the section. One way to solve this is to also apply the face to the heading prefix `* `, `** ` etc.

I also noticed that when using `*` for lists and having checkboxes, the style for checkboxes won't be applied, so I also moved these in together with the other list prefixes to make it apply.

Is there any case where you wouldn't want to align the headings?

See this for the before and after, with the regex hi-lighted to indicate where the face applies and not:

Before:
![variable-pitch-before](https://user-images.githubusercontent.com/280235/69435169-54800c00-0d3f-11ea-8a49-717aa7219d9b.png)

After:
![variable-pitch-after](https://user-images.githubusercontent.com/280235/69435191-5d70dd80-0d3f-11ea-808f-f864f974a901.png)

PS.
Commit message was a bit long when I included `org-variable-pitch-font-lock-keywords`, should I skip it, or are you okay with the message overflowing?

cadadr commented:
Hello, sorry for the very late reply.  This looks fine and I was actually thinking of adding this.  But one problem is that this causes the `org-level-*` faces to not apply to the leading asterixes.  That's because we apply a new face to matches and the other face can't be applied anymore.

To me personally, that's not really a problem, but I can't merge it like this b/c it'd break people's themes.  We either solve this problem (but IDK if that's possible) or have this be an option.  If you'll want to implement this as an optional feature, you can add a relevant defcustom toggle and use its value in `org-variable-pitch-minor-mode` to decide whether or not to also match headlines.

cadadr commented:
BTW commit message is okay, I don't care if it overflows.

terlar commented:
Thank you, I suspected that might be the case, thank you for taking that into consideration, I will make this configurable via defcustom.

cadadr commented:
You're welcome and thank you, looking forward to your patch.

> Unfortunately I have not found a way to apply several faces, I guess it could work with inherit, but then the question is which one to inherit because it could be any org-level-* face applied.

I think that's not possible because this is related to another problem we have here: we need to have links and footnotes in monospace because otherwise they mess up tables.  I haven't found a way to apply both faces (table + link/fn) reading the manual and asking on reddit.  Thus, AFAIK, multiple faces are not possible.

terlar commented:
I have updated the PR with a configurable option to enable the application of monospace face to the headline prefix. It can be configured via `org-variable-pitch-fontify-headline-prefix`. Not sure how you like the naming and the implementation or if the docstring is clear enough. Let me know if you have any suggestions for changes.

I didn't want to replicate the let-block so right now I extracted the vars outside and then do the `let` with `setq`. I also tried to do the defvar inside the let, but then the package linter complained.

cadadr commented:
Sadly the same problem persists, guess because we're trying to be a bit too clever with the regexp.  The one below is a bit more repetitive and naïve, but seems to work.  Can you please test it and use it in the patch if it works for you?

```elisp
   `((,(rx bol (1+ blank))
      ,code)
     (,(rx bol
           (1+ blank) "\*"
           (opt blank "[" (in "-X ") "]")
           blank)
      ,code)
     (,(rx bol (0+ blank)
           (or (: (or (+ digit) letter) (in ".)"))
               (: (in "-+")
                  (opt blank "[" (in "-X ") "]")))
           blank)
      ,code))
```

If you want you can move the checkbox part that's repeated into let and then quote-expand it in the rx form, but I don't mind the repetition.

terlar commented:
![image](https://user-images.githubusercontent.com/280235/70379815-20593d80-1932-11ea-83d1-e94af5c54b0e.png)

I don't see the issue, as shown in my screenshot the rule only applies to the `*` prefixed by at least one space.

What kind of issue do you see?

You can also see from the colors that the regexp doesn't apply to the headlines.

Are you perhaps using `org-bullets-mode`? Because if that is the case, it applies it's own face to the headline stars, I am mentioning this, because I ran into that when testing this.

cadadr commented:
Wow, so I somehow messed up the merge before testing, your patch works indeed.
